base/multidimensional.jl

```diff
@@ -418,6 +418,7 @@ module IteratorsMD
     # AbstractArray implementation
     Base.IndexStyle(::Type{LinearIndices{N,R}}) where {N,R} = IndexCartesian()
     Base.axes(iter::LinearIndices{N,R}) where {N,R} = iter.indices
+    Base.size(iter::LinearIndices{N,R}) where {N,R} = length.(iter.indices)
```

Comment:
At present we have the recommendation to [not implement `size` for arrays that use non-1 indexing](https://docs.julialang.org/en/latest/devdocs/offset-arrays/#Do-**not**-implement-size-or-length-1), and so perhaps you should restrict this to `OneTo` indices. But as part of #24899 (which I will tackle once we stop making breaking changes) that restriction will go away, and I for one do not object if you just keep this the way it is.

Reply:
OK. As you want, you're the expert.